src/ol/format/kmlformat.js

```diff
@@ -1162,16 +1182,15 @@ ol.format.KML.DataParser_ = function(node, objectStack) {
       'node.nodeType should be ELEMENT');
   goog.asserts.assert(node.localName == 'Data', 'localName should be Data');
   var name = node.getAttribute('name');
+  ol.xml.parseNode(ol.format.KML.DATA_PARSERS_, node, objectStack);
+  var featureObject =
+    /** @type {Object} */ (objectStack[objectStack.length - 1]);
+  goog.asserts.assert(goog.isObject(featureObject),
+      'featureObject should be an Object');
   if (name !== null) {
-    var data = ol.xml.pushParseAndPop(
-        undefined, ol.format.KML.DATA_PARSERS_, node, objectStack);
-    if (data) {
-      var featureObject =
-          /** @type {Object} */ (objectStack[objectStack.length - 1]);
-      goog.asserts.assert(goog.isObject(featureObject),
-          'featureObject should be an Object');
-      featureObject[name] = data;
-    }
+    featureObject[name] = featureObject.value;
+  } else {
+    featureObject[featureObject.displayName] = featureObject.value;
```

Comment:
If both `name` and `displayName` are given, the value of `displayName` is lost? I guess that is ok?